edb/testbase/server.py

```diff
@@ -1505,3 +1510,65 @@ def start_edgedb_server(
         compiler_pool_size=compiler_pool_size,
         debug=debug
     )
+
+
+def get_cases_by_shard(cases, current_shard, total_shards, verbosity, stats):
+    if total_shards <= 1:
+        return cases
+
+    current_shard -= 1
+    new_test_est = 0.1
+    tests_with_est = []
+    new_tests = []
+    total_tests = 0
+    selected_tests = 0
+    for tests in cases.values():
```

Comment:
I wonder if we can get better performance by avoiding the splits of test suites (classes) as much as possible.  The setup for a test suite might be expensive.  We can time and record `_setup_database` calls to guide this decision,

Reply:
Yes, I thought about that and had some early patches to group test suites by the 60+ different database setups, or leave the non-database tests in a separate group. Then I became lazy and went for an easier way to get some results first. Now that we're here, I'm totally open to more attempts.